service/dataset.py

```diff
@@ -21,25 +26,46 @@
                             "host": "https://demo.dataverse.org/api/access/datafile/395811",
                             "schema": '{"fake":"schema"}'
                         },
-                        "budget":3.0}}
+                        "budget":3.0,
+                        "authorized_users":['mock_creds']}}
 
+RELEASED_DATASETS = {"example_released_csv": {
+                        "dataset_name": "example_released_csv",
+                        "dataset_type": "csv_details",
+                        "csv_details": {
+                            "local_path": os.path.join(os.path.dirname(__file__), "datasets", "example.csv")
+                        },
+                        "budget":10.0,
+                        "authorized_users":['cb20a2f9-d97b-42b5-9e74-f9ed05b43839', 'mock_creds']},
+                    "demo_released_dataverse": {
+                        "dataset_name": "demo_released_dataverse",
+                        "dataset_type": "dataverse_details",
+                        "dataverse_details": {
+                            "local_metadata_path": os.path.join(os.path.dirname(__file__),
+                                                                "datasets",
+                                                                "dataverse",
+                                                                "demo_dataverse.yml"),
+                            "host": "https://demo.dataverse.org/api/access/datafile/395811",
+                            "schema": '{"fake":"schema"}'
+                        },
+                        "budget":10.0,
+                        "authorized_users":['mock_creds']}}
 
 KNOWN_DATASET_TYPE_KEYS = ["csv_details", "dataverse_details"]
 
-def read(dataset_request):
-    """Get information needed to load the dataset
+KNOWN_USERS = {'ab30c2f7-d97b-42b5-9e74-f9ed05b43839': defaultdict(int), 'mock_creds': defaultdict(int, {'example_released_csv': 1})}
 
-    :param info: The dataset to read and budget to use.
-    :type info: dict {"dataset_name": str, "budget":int}
-    :return: A dataset document that contains the type and info of the dataset
-    :rtype: dict{"dataset_type": str, dataset_key: dict}
-    """
+# NOTE: This could be tracked inside datasets as well, although
+# potentially safer to only track owners service side
+OWNERS = {'ab30c2f7-d97b-42b5-9e74-f9ed05b43839': ["example_csv"]}
```

Comment:
We may need owners longer term but for now an owner can be marked as having read the dataset, and therefore not incur future costs